### ISSUE TITLE
[BUG or FEATURE?] - Is MRSI supported / not available in drop-down Data scenario?

### ISSUE THREAD

**Issue by SamGuay** [schorschinho/osprey#557]
**Describe the bug**
*tl;dr: I read the docs from top to bottom, saw that MRSI was listed in the data scenario, thought it was supported. Tried to make it work with 2D fast CSI data but it hasn't so far, and errors differ between CLI and GUI.*

After having written this issue, I realize MRSI being not available in the drop-down might be more a feature than a bug as the [Supported methods in the README.md](https://github.com/schorschinho/osprey#supported-methods) doesn't mention it and seems more up to date than the [docs](https://schorschinho.github.io/osprey/the-osprey-job-file.html#data-scenario).

See additional context below for the full story.

*Initial* issue: MRSI option doesn't appear when generating a job file through the GUI. 

**To Reproduce**
1. Open `Osprey` GUI
2. Click on `Create Job`
3. Select `Data Scenario` drop-down. 
4. See only the 3 options.

**Expected behavior**
According to the [docs](https://schorschinho.github.io/osprey/the-osprey-job-file.html#data-scenario), MRSI is listed. 

**Screenshots**
![image](https://user-images.githubusercontent.com/30598330/219901599-1eb6d89b-fa80-4048-9198-592290f29e65.png)

**Computer configuration (please complete the following information):**
 - Osprey: `2.4.0`
 - OS: `Ubuntu`
 - Version: `20.04.1`
 - MATLAB version: `r2021a`

**Additional context**

1. I created a job file based on the [example job file](https://schorschinho.github.io/osprey/the-osprey-job-file.html#example-job-file) for my data, and ran into a couple of issues while using Osprey through the command-line which I thought I'd fixed. After having managed to load the data and recognized the usual FID-A output (I processed many SVS with it), I couldn't understand why the next command (`MRSCont = OspreyProcess(MRSCont)`), wouldn't process or suspect the data was wrongly loaded.

<details>
<summary> Click to see <code>OspreyLoad(MRSCont)</code> log   </summary>

```
>> MRSCont = OspreyLoad(MRSCont)                    
Timestamp February 18, 2023 14:48:59 Osprey 2.4.0  OspreyLoad
Loading raw data from dataset   1 out of   1 total datasets...
Software version: VD (!?)
Reader version: 1496740353 (UTC: 06-Jun-2017 09:12:33)
Scan 1/2, read all mdhs:
    60.7 MB read in    1 s
Scan 2/2, read all mdhs:
   207.0 MB read in    1 s

ans =

     0

multi RAID file detected.
Software version: VD (!?)
Reader version: 1496740353 (UTC: 06-Jun-2017 09:12:33)
Scan 1/2, read all mdhs:
    60.7 MB read in    1 s
Scan 2/2, read all mdhs:
    42.1 MB read in    0 s

ans =

     0

multi RAID file detected.

... done.
 Elapsed time 25.574929 seconds

MRSCont = 

  struct with fields:

           flags: [1x1 struct]
            opts: [1x1 struct]
       ospFolder: '/data/beaumont/software/osprey'
           files: {'/home/user/MRS_project/sub-014/ses-pre/mrs/sub-014_ses-pre_csi-se-30.dat'}
        files_mm: {}
       files_ref: {'/home/user/MRS_project/sub-014/ses-pre/mrs/sub-014_ses-pre_csi-se-30_w.dat'}
         files_w: {}
    files_mm_ref: {}
       files_nii: {'/home/user/MRS_project/sub-014/ses-pre/anat/sub-014/ses-pre_T1w.nii.gz'}
       files_seg: {}
     files_sense: {}
    outputFolder: '/home/user/MRS_project/derivatives'
        colormap: [1x1 struct]
       loadedJob: 'MRS_project/code/jobfile.m'
             ver: [1x1 struct]
      outputFile: 'jobfile.mat'
       nDatasets: [1 1]
          vendor: 'Siemens'
        datatype: 'TWIX'
             raw: {[1x1 struct]}
         raw_ref: {[1x1 struct]}
         runtime: [1x1 struct]
            plot: [1x1 struct]
```
</details>

<details>
<summary> Click to see <code>OspreyProcess(MRSCont)</code> error log   </summary>

```
>> MRSCont = OspreyProcess(MRSCont)                 
Timestamp February 18, 2023 14:50:24 Osprey 2.4.0  OspreyProcess
Processing data from dataset   1 out of   1 total datasets...
Unrecognized field name "names".

Error in osp_XReferencing (line 42)
    ind = find(strcmp(dataToFit.names,'sum'));

Error in op_preref (line 176)
            [refShift, ~] = osp_XReferencing(temp_spec,frequencies,polarity,lim,0);% determine
            frequency shift

Error in OspreyProcess (line 214)
                 [refShift_ind_ini]=op_preref(raw,seq);
 
>> 
```
</details>

2. Then, I tried using the GUI to see if the errors would be more verbose using the same job file. 

<details>
<summary> Click to see <code>GUI load (and plot)</code> error log </summary>

```
>> Osprey
Timestamp February 18, 2023 18:18:01 Osprey 2.4.0  OspreyGUI
Warning: MATLAB has disabled some advanced graphics rendering features by switching to software OpenGL. For more
information, click here. 
Continue with loading new job, overwriting existing job.
/home/user/MRS_project/code/jobfile.m
Timestamp February 18, 2023 18:19:17 Osprey 2.4.0
Timestamp February 18, 2023 18:19:17 Osprey 2.4.0  OspreyGUI
Timestamp February 18, 2023 18:20:33 Osprey 2.4.0  OspreyLoad
Loading raw data from dataset   1 out of   1 total datasets...
Software version: VD (!?)
Reader version: 1496740353 (UTC: 06-Jun-2017 09:12:33)
Scan 1/2, read all mdhs:
    60.7 MB read in   56 s
Scan 2/2, read all mdhs:
   207.0 MB read in   46 s

ans =

     0

multi RAID file detected.
Software version: VD (!?)
Reader version: 1496740353 (UTC: 06-Jun-2017 09:12:33)
Scan 1/2, read all mdhs:
    60.7 MB read in   55 s
Scan 2/2, read all mdhs:
    42.1 MB read in   45 s

ans =

     0

multi RAID file detected.

... done.
 Elapsed time 223.900988 seconds
Error using plot
Data cannot have more than 2 dimensions.

Error in osp_plotLoad (line 285)
            plot(axesHandles.A, dataToPlot.ppm, real(dataToPlot.specs(:,rr)) + rr*stag, 'k', 'LineWidth',
            0.5, 'Color', MRSCont.colormap.Foreground);

Error in osp_iniLoadWindow (line 418)
                temp = osp_plotLoad(MRSCont, gui.controls.Selected,'ref'); %re_mm

Error in osp_onLoad (line 66)
    osp_iniLoadWindow(gui);
 
Error while evaluating UIControl Callback.
```
</details>

After getting this error, I get a blank page in the Metabolite tab and see image below for the reference tab. It seems to output the sequence detail but that is all. 

![image](https://user-images.githubusercontent.com/30598330/219903929-21eb454a-f48d-454e-994d-a53a6cd3f300.png)

3. So, I thought of generating a new job file through job-file-generator GUI to see if *I* was the problem but quickly realized there was no MRSI option. I did try all the options, but the same error happens.

4. I started digging even more into the docs/code to find hints of it being supported or not for siemens data as I've seen the `OspreyQuantifyMRSI` function. Though, it may not have been implemented for Siemens data, yet.

So in the end, should I look into another software or help implementing the feature? (Im very not a matlab developer but I can try..!)

Maybe I have completely missed a detail or a discussion on MRShub or something that would save my life... I'm open to other suggestions! :)

Thanks

**Comment by HJZollner** [comment]
Hi @SamGuay,

Thank you for taking interest in Osprey and reaching out. Also this is a remarkably detailed bug report - very much appreciated!

As you have realized there are several hints towards MRSI processing in the current develop branch and Osprey release. However, they are not fully functional and especially not integrated in the GUI - hence not visible in the dropdown menu.

If you are interested in MRSI processing with Osprey you should check out (https://github.com/schorschinho/osprey/tree/MRSI) which is better developed. However, it only supports Philips data/list or Philips sdat/spar and unfortunately NOT Siemens TWIX files.

Depending on what your timeline and project goals are I'd highly recommend looking into FSL-MRS (https://open.win.ox.ac.uk/pages/fsl/fsl_mrs/#)  or ABFit (https://github.com/martin3141/spant) which are both capable of analyzing Siemens CSI data (after nifti-mrs conversion  https://github.com/wtclarke/mrs_nifti_standard).

Or if you are interested in implementing/contributing to Osprey you could fix the twix_loader in Osprey to correctly identify and sort MRSI data for further processing. I am happy to share some example files as reference.

Best,
Helge

**Comment by SamGuay** [comment]
Hi! Thanks for taking the time to reply! Last month was pretty busy with other stuff but I started working on implementing the nifti-mrs conversion for TWIX as a start since I'm BIDSifying all my data. Once I have my data standardized, I might come back to Osprey and dig a little deeper into the twix_loader.
